crates/ruff_python_formatter/src/options.rs

```diff
@@ -349,3 +364,20 @@ where
         )),
     }
 }
+
+#[derive(CacheKey, Clone, Copy, Debug, PartialOrd, Ord, PartialEq, Eq, Default)]
+#[cfg_attr(
+    feature = "serde",
+    derive(serde::Serialize, serde::Deserialize),
+    serde(rename_all = "lowercase")
+)]
+#[cfg_attr(feature = "schemars", derive(schemars::JsonSchema))]
+pub enum PythonVersion {
+    Py37,
+    #[default]
+    Py38,
+    Py39,
+    Py310,
+    Py311,
+    Py312,
+}
```

Comment:
It would be nice if we could have a single `PythonVersion` definitions shared by the linter and formatter but there's currently no good crate where to place it. Creating one feels overkill for such a simple definition. 

Ultimately I think the definition should be in `ruff_python_ast` where it specifies the language features.